cmd/update.go

```diff
@@ -0,0 +1,197 @@
+
+	executablePath, err := os.Executable()
+	currentDir := filepath.Dir(executablePath)
+
+	if err != nil {
+		panic("Error getting current executable path")
+	}
+	updatedBytes, err := os.ReadFile(fileName)
+	if err != nil {
+		panic(err)
+	}
+	tempBinaryPath := fmt.Sprintf("%s/updatedGrype", currentDir)
+	err = os.WriteFile(tempBinaryPath, updatedBytes, os.FileMode(0755))
+	if err != nil {
+		panic(err)
+	}
+	err = os.Rename(tempBinaryPath, executablePath)
+	if err != nil {
+		log.Errorf("Error while overwriting current executable: %s", err)
+	}
+	fmt.Println("Grype updated successfully! Run 'grype version' to get more info")
+	log.Infof("Updated grype (%s) successfully\n", executablePath)
+}
+
+func ExtractTarGz(gzipStream io.Reader, destination string) error {
+	uncompressedStream, err := gzip.NewReader(gzipStream)
+	if err != nil {
+		log.Error("ExtractTarGz: NewReader failed")
+		return err
+	}
+
+	tarReader := tar.NewReader(uncompressedStream)
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			log.Errorf("ExtractTarGz: Next() failed: %s", err.Error())
+			return err
+		}
+
+		path := fmt.Sprintf("%s/%s", destination, header.Name)
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.Mkdir(path, 0755); err != nil {
+				log.Errorf("ExtractTarGz: Mkdir() failed: %s", err.Error())
+				return err
+			}
+		case tar.TypeReg:
+			outFile, err := os.Create(path)
+			if err != nil {
+				log.Errorf("ExtractTarGz: Create() failed: %s", err.Error())
+				return err
+			}
+			for {
+				_, err := io.CopyN(outFile, tarReader, 1024)
+				if err != nil {
+					if err == io.EOF {
+						break
+					}
+					log.Errorf("ExtractTarGz: Copy() failed: %s", err.Error())
+					return err
+				}
+			}
+			outFile.Close()
+
+		default:
+			log.Errorf(
+				"ExtractTarGz: uknown type: %s in %s",
+				header.Typeflag,
+				header.Name)
+			return err
+		}
+	}
+	return nil
+}
+
+func checkLatestVersion() (bool, string, error) {
+	updateAvaliable, version, err := version.IsUpdateAvailable()
+	if err != nil {
+		log.Errorf("ERROR")
+		fmt.Println(err)
+		return false, "", err
+	}
+	if updateAvaliable {
+		log.Infof("New version available v%s", version)
+		return true, version, nil
+	}
+	log.Infof("No updates available")
+	return false, "", nil
+}
+
+func downloadCompressed(version string) string {
+	var extension = "tar.gz"
+	if runtime.GOOS == "windows" {
+		extension = "zip"
+	}
+	f, err := os.CreateTemp("", "grype")
+	log.Debugf("Created temporary file %s", f.Name())
+	if err != nil {
+		defer os.Remove(f.Name())
+		panic(err)
+	}
+
+	var downloadURLTemplate = "https://github.com/%s/%s/releases/download/v%s/grype_%s_%s_%s.%s"
+	var downloadURL = fmt.Sprintf(downloadURLTemplate, "anchore", "grype", version, version, runtime.GOOS, runtime.GOARCH, extension)
+	log.Infof("Downloading v%s", version)
+	fmt.Printf("Downloading v%s...\n", version)
+	resp, err := http.Get(downloadURL) //nolint
```

Comment:
Added `//nolint` here to address `G107: Url provided to HTTP request as taint input` (https://securego.io/docs/rules/g107.html). Is there a more elegant way to resolve this?

Reply:
Just adding context here to make sure users know why this would not be considered a tainted input. In this case the user has control over some parts of the URL string but would have to hijack the full domain for the redirect to happen.

Consider also formatting the string and using https://pkg.go.dev/net/url#Parse to parse the input into a valid URL struct